src/KExtreme.chpl

```diff
@@ -0,0 +1,110 @@
+/*
+ * kextreme data structure that is used
+ * to track the minimum or maximum `size`
+ * values that are in an array. Acts similar
+ * to a heap data structure, but can be
+ * sorted and merged to extract the
+ * extreme values from multiple kextreme
+ * objects.
+ */
+
+module KExtreme {
+  use SymArrayDmap;
+  use Sort;
+  record kextreme {
```

Comment:
i'd like a little more documentation describing the record and maybe how it gets used with the chapel reduction framework, also since it's a definition I think Id like it to follow the rest of the code and use uppercase camel case, I know that would make it collide with the module name... maybe rename the module.

Reply:
it would also be good to make the doc chpldoc/sphinx formatted.